### PR TITLE
Notes view autosave no longer removes changes on notes being edited

### PR DESCRIPTION
**Pull request by altair1673**
### Prerequisites

- [x] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

**_Fix for issue #1848 found during testing of release candidate v2.10.0-rc1_**

Notes that are being edit do not get reset when an autosave occurs. If a note is in edit mode and the user switches graphs, that note will remain in edit mode with any changes made when the user returns.

### Why Should This Be In Core?

Makes editing notes safe as the users can the sure there changes won't randomly be lost if a background plugin finishes running.
### Benefits

Increases usability of the notes view.
### Possible Drawbacks

Auto notes are not updated in real time.
### Verification Process

Made changes to note in edit mode switched graphs and came back to see note still in edit mode with the changes I have made.
### Applicable Issues

#1848

### CODE REVIEW

File: CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/state/NotesViewEntry.java [hunk 1]
```diff
@@ -149,6 +153,31 @@ public void setID(final int id) {
 
     }
 
+    public boolean checkIfWasInEditMode() {
+        return wasInEditMode;
+    }
+
+    public void setWasInEditMode(boolean wasInEditMode) {
```
Comment (OrionsGuardian) [inline CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/state/NotesViewEntry.java:160]: final

File: CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/state/NotesViewEntry.java [hunk 2]
```diff
@@ -149,6 +153,31 @@ public void setID(final int id) {
...
+        this.wasInEditMode = wasInEditMode;
+    }
+
+    public String getTempContent() {
+        return tempContent;
+    }
+
+    public void setTempContent(String tempContent) {
```
Comment (OrionsGuardian) [inline CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/state/NotesViewEntry.java:168]: final

File: CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/state/NotesViewEntry.java [hunk 3]
```diff
@@ -149,6 +153,31 @@ public void setID(final int id) {
...
+        this.tempContent = tempContent;
+    }
+
+    public String getTempTitle() {
+        return tempTitle;
+    }
+
+    public void setTempTitle(String tempTitle) {
```
Comment (OrionsGuardian) [inline CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/state/NotesViewEntry.java:176]: final

**Review by OrionsGuardian — CHANGES_REQUESTED** [review]

**Review by Delphinus8821 — APPROVED** [review]

**Review by OrionsGuardian — APPROVED** [review]

File: CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/NotesViewPane.java [hunk 4]
```diff
@@ -583,16 +583,29 @@ protected synchronized void updateNotesUI() {
         }
         
         Platform.runLater(() -> {
-            notesListVBox.getChildren().removeAll(notesListVBox.getChildren());
+            boolean foundNoteInEdit = false;
+            for (int i = 0; i < notesToRender.size(); i++) {
```
Comment (antares1470) [inline CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/NotesViewPane.java:587]: use a for each here `for (final NotesViewEntry note : notesToRender)`

File: CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/state/NotesViewEntry.java [hunk 5]
```diff
@@ -40,8 +40,12 @@ public class NotesViewEntry implements PluginReportListener {
     private List<Integer> transactionsSelected;
     private List<String> tags = new ArrayList<>();
     private boolean editMode;
+    private boolean wasInEditMode = false;
     private boolean isShowing = true;
 
+    private String tempContent = "";
+    private String tempTitle = "";
```
Comment (antares1470) [inline CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/state/NotesViewEntry.java:47]: add the instantiations to the constructor

**Review by antares1470 — CHANGES_REQUESTED** [review]

**Review by antares1470 — APPROVED** [review]
Good to go once the merge conflict is resolved

**Pull request merged**